src/test/java/org/sagebionetworks/bridge/spring/controllers/CRCControllerTest.java

```diff
@@ -0,0 +1,665 @@
+        mockRequestBody(mockRequest, makeObservation());
+        
+        DateRangeResourceList<? extends ReportData> results = new DateRangeResourceList<>(ImmutableList.of());
+        doReturn(results).when(mockReportService).getParticipantReport(
+                APP_ID, OBSERVATION_REPORT, HEALTH_CODE, JAN1, JAN2);
+        
+        ResponseEntity<StatusMessage> retValue = controller.postObservation();
+        assertEquals(retValue.getBody().getMessage(), "Observation created.");
+        assertEquals(retValue.getStatusCodeValue(), 201);
+        
+        verify(mockAccountService).authenticate(eq(app), signInCaptor.capture());
+        SignIn capturedSignIn = signInCaptor.getValue();
+        assertEquals(capturedSignIn.getAppId(), APP_ID);
+        assertEquals(capturedSignIn.getExternalId(), USERNAME);
+        assertEquals(capturedSignIn.getPassword(), "dummy-password");
+        
+        verify(mockReportService).saveParticipantReport(eq(APP_ID), eq(OBSERVATION_REPORT), eq(HEALTH_CODE),
+                reportCaptor.capture());
+        ReportData capturedReport = reportCaptor.getValue();
+        assertEquals(capturedReport.getDate(), "1970-01-01");
+        verifyIdentifier(capturedReport.getData());
+        assertEquals(capturedReport.getSubstudyIds(), USER_SUBSTUDY_IDS);
+        
+        verify(mockAccountService).updateAccount(accountCaptor.capture(), isNull());
+        Account capturedAcct = accountCaptor.getValue();
+        assertEquals(capturedAcct.getDataGroups(), makeSetOf(CRCController.AccountStates.TESTS_AVAILABLE));
+        assertEquals(capturedAcct.getAttributes().get(TIMESTAMP_FIELD), TIMESTAMP.toString());
+    }
+    
+    @Test
+    public void postObservationUpdated() throws Exception {
+        when(mockRequest.getHeader(AUTHORIZATION)).thenReturn(AUTHORIZATION_HEADER_VALUE);
+        when(mockAccountService.authenticate(any(), any())).thenReturn(account);
+        when(mockAccountService.getAccount(ACCOUNT_ID)).thenReturn(account);
+        mockRequestBody(mockRequest, makeObservation());
+        
+        DateRangeResourceList<? extends ReportData> results = new DateRangeResourceList<>(ImmutableList.of(ReportData.create()));
+        doReturn(results).when(mockReportService).getParticipantReport(
+                APP_ID, OBSERVATION_REPORT, HEALTH_CODE, JAN1, JAN2);
+        
+        ResponseEntity<StatusMessage> retValue = controller.postObservation();
+        assertEquals(retValue.getBody().getMessage(), "Observation updated.");
+        assertEquals(retValue.getStatusCodeValue(), 200);
+    }
+
+    @Test
+    public void authenticationWorks() {
+        when(mockRequest.getHeader(AUTHORIZATION)).thenReturn(AUTHORIZATION_HEADER_VALUE);
+        when(mockAppService.getApp(CRCController.APP_ID)).thenReturn(mockApp);
+        when(mockAccountService.authenticate(eq(mockApp), signInCaptor.capture())).thenReturn(account);
+        
+        controller.httpBasicAuthentication();
```

Comment:
Do we want to check that substudies are propagated properly?

Reply:
I'm not sure what you mean here...